packages/bee-q/src/components/dialog/bq-dialog.tsx

```diff
@@ -0,0 +1,152 @@
+import { h, Component, Prop, Element, Watch, State, Method, Host } from '@stencil/core';
+
+import { DIALOG_SIZE, DIALOG_FOOTER_VARIANT, TDialogSize, TDialogFooterVariant } from './bq-dialog.types';
+import { validatePropValue } from '../../shared/utils';
+
+/**
+ * @part base - The component wrapper container inside the shadow DOM
+ * @part container - The `<div>` container that holds the dialog content
+ * @part button-close - The button that close the dialog on click
+ */
+
+@Component({
+  tag: 'bq-dialog',
+  styleUrl: './scss/bq-dialog.scss',
+  shadow: true,
+})
+export class BqDialog {
+  // Own Properties
+  // ====================
+
+  private overlayElem: HTMLDivElement;
+
+  // Reference to host HTML element
+  // ===================================
+  @Element() el!: HTMLBqDialogElement;
+
+  // State() variables
+  @State() isOpen = false;
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** The size of the dialog */
+  @Prop({ reflect: true, mutable: true }) size: TDialogSize = 'large';
```

Comment:
I think the default value in terms of size for any component is `medium`.

Reply:
Done